### ISSUE TITLE
Code for calculating Pre-training labels is missing

### ISSUE THREAD

**Issue by ArjunDosajh** [hspark1212/MOFTransformer#159]
I couldn't find the code for calculating the pre-training task labels anywhere in the repository. These tasks include mtp, vfp and moc. In the paper it was mentioned that some of these calculations were done using Zeo++. It would be very helpful if code for this can be provided, since pre-training on new datasets won't be possible without this.

**Comment by hspark1212** [comment]
Hi @ArjunDosajh, as you mentioned, we calculated void fraction of pre-training dataset using Zeo++. I'm wondering if you need either a running script for ZEO++ or pre-traning code for mtp, vfp, moc.

**Comment by ArjunDosajh** [comment]
I require a script which takes input a directory path (this directory contains all the CIF files) and creates json files for mtp, vfp and moc using ZEO++. It would be very helpful if you could provide the script so that I can directly prepare everything necessary for pre-training on some other dataset.

**Comment by Yeonghun1675** [comment]
Hi @ArjunDosajh,

We used ZEO++ to create the labels only for vfp. For moc and mtp labels, the code is based on the assumption that the structure was created with pormake (https://github.com/Yeonghun1675/bulk_pormake_generation).

Is your dataset created using pormake? If not, unfortunately you will have to create new code for your dataset.

For the ZEO++ code for VFP, it's uploaded to github: (https://github.com/Yeonghun1675/zeo-) It's still unorganized, so I'll get back to you after I clean it up.

**Comment by ArjunDosajh** [comment]
Hi @Yeonghun1675,
My dataset is not created using pormake. I'll try to write the code for MTP and MOC labels on my own then. Anyways, thanks for the help!
